app/renderer/ui/containers/WalletForm.tsx

```diff
@@ -0,0 +1,393 @@
+import { saveWallet } from "app/renderer/actions/loginForm"
+
+import { WalletForm } from "app/renderer/ui/components/walletForm"
+import {
+  AddressesGeneration,
+  WalletEncryptionPassword,
+  WalletSeedBackup,
+  WalletSeedConfirmation,
+  WalletSeedTypeSelection,
+  Welcome
+} from "app/renderer/ui/components/walletForm/steps"
+
+/**
+ * Props that match redux store state
+ *
+ * @export
+ * @interface StateProps
+ */
+export interface StateProps {
+  wallets: Wallets,
+}
+
+/**
+ * Props that match redux actions
+ *
+ * @export
+ * @interface DispatchProps
+ */
+export interface DispatchProps {
+  actions: any
+}
+
+/**
+ * Props that contain configuration
+ *
+ * @interface Props
+ */
+interface Props {
+  services: any
+}
+
+/**
+ * Function to map the dispatch of actions to props
+ * @param dispatch
+ */
+const mapDispatchToProps = (dispatch: Dispatch<IAction>): DispatchProps => {
+  return {
+    actions: bindActionCreators({ saveWallet }, dispatch)
+  }
+}
+
+/**
+ * Function to map store state to props
+ * @param state
+ */
+const mapStateToProps = (state: StoreState): StateProps => {
+  return ({
+    wallets: state.wallets,
+  })
+}
+
+export const PATHS = {
+  WELCOME: "/signup/welcome",
+  WALLET_SEED_TYPE_SELECTION: "/signup/wallet-seed-type-selection",
+  WALLET_SEED_BACKUP: "/signup/wallet-seed-backup",
+  WALLET_SEED_CONFIRMATION: "/signup/wallet-seed-confirmation",
+  WALLET_ENCRYPTION_PASSWORD: "/signup/wallet-encryption-password",
+  ADDRESS_GENERATION: "/signup/address-generation",
+  MAIN: "/main"
+}
+
+/**
+ * Wallet form container UI component
+ *
+ * @class WalletFormContainer
+ * @extends {(React.Component<RouteComponentProps<any> & Props & StateProps & DispatchProps>)}
+ */
+class WalletFormContainer extends
+  React.Component<RouteComponentProps<any> & Props & StateProps & DispatchProps> {
+
+  /**
+   * component state
+   *
+   * @memberof WalletFormContainer
+   */
+  public state = {
+    confirmMnemonics: "",
+    mnemonics: "",
+    mnemonicsErrorMessage: "",
+    password: "",
+    passwordErrorMessage: "",
+    repeatPassword: "",
+    spinner: false,
+  }
+
+  /**
+   * method to handle react router navigation
+   *
+   * @private
+   * @memberof WalletFormContainer
+   */
+  private to = (string: string) => () => { this.props.history.replace(string) }
+
+  /**
+   * method to render welcome step
+   *
+   * @private
+   * @memberof WalletFormContainer
+   */
+  private renderWelcome = () => <Welcome nextStep={this.to(PATHS.WALLET_SEED_TYPE_SELECTION)} />
+
+  /**
+   * method to render seed type selection step
+   *
+   * @private
+   * @memberof WalletFormContainer
+   */
+  private renderSeedTypeSelection = () => {
+    const steps = {
+      // TODO: add seed type options
+      newSeed: async () => {
+        if (!this.state.mnemonics) {
+          this.setState({ spinner: true })
+          const mnemonicsResponse = await newMnemonics(this.props.services.apiClient)
```

Comment:
I would prefer to use Redux to handle API communication, problem is that here we are not handling the case where the backend fails with an exception (or panic) which will leave the frontend in an inconsistent state and not responsive leaving the user with a _uh_ face

Reply:
IMO, RPC level and JSON-RPC level errors should be handled by the API client methods.

[Here's an example](https://github.com/witnet/sheikah/blob/master/app/renderer/api/getWallet.ts#L16) by @mmartinbar and @mariocao

Reply:
Then it's up to the caller of methods like `newMnemonics()` to handle all cases in an exhaustive way (with `.then` and `.catch` if possible, as `await` is a ticking time-bomb when used in the frontend :rofl: )